### ISSUE TITLE
"show decompiled" / "show semanticdb" etc. and split view

### ISSUE THREAD

**Issue by kubukoz** [scalameta/metals-vscode#1131]
**Describe the bug**

When using a split view, using a command like "Show decompiled with javap" or "show semanticdb" may use the unexpected file (as in, not the one you currently have a cursor on).

It seems like Metals has a heuristic for picking the file based on the last edit / last opened file. I assume this is because `workspace/executeCommand` doesn't provide the currently focused file (it's not a `textDocument` request, after all).

**To Reproduce** Steps to reproduce the behavior:

1. Create a couple files in a workspace
2. Open two or more of them in a split pane view
3. Edit file A
4. Focus on file B
5. Ask metals to show semanticdb

**Expected behavior**

File B's semanticdb is shown

**Screenshots**

<img width="1687" alt="image" src="https://user-images.githubusercontent.com/894884/187085512-da1d5837-34cf-461f-af10-6198826d5b3a.png">

(make an edit in B)

<img width="1149" alt="image" src="https://user-images.githubusercontent.com/894884/187085521-eee227cf-b102-485e-84e0-7aea9145a998.png">

(click on A, ask for semanticdb)

<img width="1294" alt="image" src="https://user-images.githubusercontent.com/894884/187085533-548b0dd3-2e78-4d38-8aea-71a757488613.png">

**Installation:**

- Operating system: macOS
- VSCode version: 1.70.2
- VSCode extension version: 1.19.7
- Metals version: 0.11.8+38-1832a3cf-SNAPSHOT

**Additional context**

**Search terms**

split pane, split view, executeCommand, command

**Comment by kubukoz** [comment]
I don't really know of a workaround except for a client-side `registerTextEditorCommand` - but I know it's a thing that'll have to be implemented by each language client. 

A long-term solution would probably be to suggest adding a `textDocument/executeCommand` in LSP.

**Comment by kpodsiad** [comment]
Current heuristic - find first matching editor (comment apparently lies)
https://github.com/scalameta/metals-vscode/blob/2954bcac826888da34f1778bb3d0536ed35e5778/src/metalsContentProvider.ts#L63-L76

Simple workaround - show all matching editors and let user pick one with quickpick.

**Comment by kubukoz** [comment]
yeah, in the case of javap there's already a selection required (which class) - should there be another one or should it be a concatenation of each file's classes?

**Comment by kpodsiad** [comment]
> yeah, in the case of javap there's already a selection required (which class) - should there be another one or should it be a concatenation of each file's classes?

Similar mechanism, but this time on frontend will be needed. Server accepts pair (uri, "decompile" kind (tasty, semanticDB, javap)) and it shows quickpick for cases when there are more eligible classes in given file. 

In this case, quickpick can be shown when there are more matching editors, but this can be done solely on the frontend side.

> **To Reproduce** Steps to reproduce the behavior:
> 
> 1. Create a couple files in a workspace
> 2. Open two or more of them in a split pane view
> 3. Edit file A
> 4. Focus on file B
> 5. Ask metals to show semanticdb

BTW, I can't reproduce it, I'm always getting semanticDB view for currently focused file 🤔
